esp/services/ws_smc/ws_smcService.cpp

```diff
@@ -2200,6 +2200,82 @@ inline const char *controlCmdMessage(int cmd)
     return NULL;
 }
 
+bool CWsSMCEx::onRoxieXrefCmd(IEspContext &context, IEspRoxieXrefCmdRequest &req, IEspRoxieXrefCmdResponse &resp)
+{
+    context.ensureFeatureAccess(ROXIE_CONTROL_URL, SecAccess_Full, ECLWATCH_SMC_ACCESS_DENIED, SMC_ACCESS_DENIED);
+
+    StringBuffer controlReq;
+    if (0==req.getQueryIds().length())
+        controlReq.append("<control:getQueryXrefInfo/>");
+    else
+    {
+        controlReq.append("<control:getQueryXrefInfo>");
+        ForEachItemIn(i, req.getQueryIds())
+        {
+            const char *id = req.getQueryIds().item(i);
+            if (!isEmptyString(id))
+                controlReq.appendf("<Query id='%s'/>", id);
+        }
+        controlReq.append("</control:getQueryXrefInfo>");
+    }
+
+#ifndef _CONTAINERIZED
+    const char *process = req.getRoxieCluster();
```

Comment:
minor: This code shares quite a few similarities with other code in this file (onRoxieControlCmd()) and ws_workunitQuerySets.cpp.  I suspect it could be simplified by combining them.